packages/taquito-sapling/src/sapling-transaction-viewer.ts

```diff
@@ -1,23 +1,275 @@
+   * The response properties are in Uint8Array format; use the getIncomingAnd Outgoing Transactions method for readable properties
+   *
+   * @param saplingId sapling id that can be found in the contract storage
+   *
+   */
+  async getIncomingAndOutgoingTransactionsRaw(saplingId: string) {
+    const incoming = [];
+    const outgoing = [];
+    const { commitments_and_ciphertexts, nullifiers } = await this.#rpcClient.getSaplingDiffById(
+      saplingId
+    );
+    for (let i = 0; i < commitments_and_ciphertexts.length; i++) {
+      const decryptedAsReceiver = await this.decryptCiphertextAsReceiver(
+        commitments_and_ciphertexts[i]
+      );
+      const decryptedAsSender = await this.decryptCiphertextAsSender(
+        commitments_and_ciphertexts[i]
+      );
+      if (decryptedAsReceiver) {
+        const balance = convertValueToBigNumber(decryptedAsReceiver.value);
+        const isSpent = await this.isSpent(
+          decryptedAsReceiver.paymentAddress,
+          balance.toString(),
+          decryptedAsReceiver.rcm,
+          i,
+          nullifiers
+        );
+        incoming.push({ ...decryptedAsReceiver, isSpent });
+      }
+      if (decryptedAsSender) {
+        outgoing.push(decryptedAsSender);
+      }
+    }
+    return {
+      incoming,
+      outgoing,
+    };
+  }
+
+  /**
+   * @description Retrieve all the incoming and outgoing decoded transactions associated with the configured viewing key
+   *
+   * @param saplingId sapling id that can be found in the contract storage
+   *
+   */
+  async getIncomingAndOutgoingTransactions(
+    saplingId: string
+  ): Promise<SaplingIncomingAndOutgoingTransaction> {
+    const tx = await this.getIncomingAndOutgoingTransactionsRaw(saplingId);
+    const incoming = tx.incoming.map(({ isSpent, ...rest }) => {
+      return { ...readableFormat(rest), isSpent };
+    });
+    const outgoing = tx.outgoing.map((outgoingTx) => {
+      return readableFormat(outgoingTx);
+    });
+    return { incoming, outgoing };
+  }
 
-    getViewingKey(){
-        return this.#viewingKey;
+  private async decryptCiphertextAsReceiver(
+    commitmentsAndCiphertexts: CommitmentsAndCiphertexts
+  ): Promise<SaplingTransactionPropertiesRaw | undefined> {
+    const commitment = commitmentsAndCiphertexts[0];
+    const { epk, payload_enc, nonce_enc } = commitmentsAndCiphertexts[1];
+
+    const ivk = await this.#viewingKeyProvider.getIncomingViewingKey();
+    const keyAgreement = await sapling.keyAgreement(epk, ivk);
+    const keyAgreementHash = blake.blake2b(keyAgreement, Buffer.from(KDF_KEY), 32);
+
+    const decrypted = await this.decryptCiphertext(
+      keyAgreementHash,
+      hex2buf(nonce_enc),
+      hex2buf(payload_enc)
+    );
+
+    if (decrypted) {
+      const { diversifier, value, rcm, memo } = this.extractTransactionProperties(decrypted);
+      const paymentAddress = bufToUint8Array(
+        await sapling.getRawPaymentAddressFromIncomingViewingKey(ivk, diversifier)
```

Comment:
what do you think about shortening IncomingViewingKey and OutgoingViewingKey in the method names to IVK and OVK?

Reply:
I can make the change yes, they will be shorter.